openstates/in/bills.py

```diff
@@ -45,198 +45,185 @@ class INBillScraper(BillScraper):
-            for bill_id, bill_url, bill_title in subject_bills:
-                chamber = 'upper' if bill_id[0] == 'S' else 'lower'
-                self.scrape_bill(
-                    chamber, term, bill_id, bill_url, bill_title, subject=subject)
-                seen_bill_ids.add(bill_id)
-
-        # Then hit bill index page to catch any uncategorized bills.
-        uls = doc.xpath('//ul[contains(@class, "clean-list")]')
-
-        for chamber, ul in zip(chambers, uls):
-            for li in ul.xpath('li'):
-                bill_id = li.xpath('string(a/strong)')
-                if bill_id in seen_bill_ids:
-                    continue
-                bill_url = li.xpath('string(a/@href)')
-                try:
-                    bill_title = li.xpath('a/strong')[0].tail.rstrip().lstrip(': ')
-                except IndexError:
-                    continue
+    def make_html_source(self,session,bill_id):
+        url = "http://iga.in.gov/legislative/{}/".format(session)
+        urls = {
+            "hb":"bills/house/",
+            "hr":"resolutions/house/simple/",
+            "hcr":"resolutions/house/concurrent",
+            "hjr":"resolutions/house/joint",
+            "hc":"resolutions/house/concurrent",
+            "hj":"resolutions/house/joint",
+            "sb":"bills/senate/",
+            "sr":"resolutions/senate/simple/",
+            "scr":"resolutions/senate/concurrent",
+            "sjr":"resolutions/senate/joint",
+            "sc":"resolutions/senate/concurrent",
+            "sj":"resolutions/senate/joint"
+            }
+        bill_id = bill_id.lower()
+        try:
+            int(bill_id[2])
+        except ValueError:
+            bill_prefix = bill_id[:3]
+            bill_num = bill_id[3:]
+        else:
+            bill_prefix = bill_id[:2]
+            bill_num = bill_id[2:]
 
-                if bill_title.strip() == "":
+        try:
+            url += urls[bill_prefix]
+        except KeyError:
+            raise AssertionError("Unknown bill type {}, don't know how to make url".format(bill_id))
+        url += bill_num
+        return url
+
+
+
+    def get_name(self,random_json):
+        #got sick of doing this everywhere
+        return random_json["firstName"] + " " + random_json["lastName"]
+
+    def scrape(self, session, chambers):
+        
+        api_base_url = "https://api.iga.in.gov"
+        client = ApiClient(self)
+        r = client.get("bills",session=session)
+        all_pages = client.unpaginate(r)
+        for b in all_pages:
+            bill_id = b["billName"]
+            bill_link = b["link"]
+            api_source = api_base_url + bill_link
+            bill_json = client.get("bill",session=session,bill_id=bill_id.lower())
+            
+            title = bill_json["title"]
+            #sometimes title is blank
+            #if that's the case, we can check to see if
+            #the latest version has a short description
+            if not title:
```

Comment:
I'm seeing some cases of `title` being `"NoneNone"`, but I'm not sure what's causing this. Check out HC 15 for an example.